dashboard/test/ui/features/step_definitions/steps.rb

```diff
@@ -1145,11 +1145,18 @@ def convert_keys(keys)
   keys.chars.map {|k| k == "\n" ? :enter : k}
 end
 
-# Known issue: IE does not register the key presses in this step.
-# Add @no_ie tag to your scenario to skip IE when using this step.
 And(/^I press keys "([^"]*)" for element "([^"]*)"$/) do |key, selector|
   element = @browser.find_element(:css, selector)
   press_keys(element, key)
+  only_alphanumeric_backslash = key.gsub(/[^0-9a-z \\]/i, '') == key
+  check_key_values = only_alphanumeric_backslash && selector != ".ace_text-input"
+  if check_key_values
+    wait_short_until do
+      element_text = element.attribute("value")
+      input_key = key.delete "\n"
+      element_text.include? input_key
+    end
+  end
```

Comment:
Just to capture our discussion: 

For a newcomer arriving at this code, even if a lot of comments are added, it feels like there is a lot of special-casing here, mostly related to Droplet, that will be hard to figure out.  I think it would probably be better to keep "I press keys X for element Y" really simple, where it just does what it says, which is press some specific keys.  That means going back to how it was before.

This would mean that if we want to wait until we know the text has been entered, we would add another step that does that, and modify the tests to call that additional step.  The downside is that we don't get the wait everywhere "for free" without changing the tests, but the upside is that we can customise what we look for to handle specific situations, such as in Droplet.

One other possibility would be to have two different steps here: one which is the regular "I press keys X for element Y" and does wait for the result in a typical input, and another which is a "I press keys X for element Y in Droplet" and which does the special-casing unique to Droplet.

Reply:
Thanks Brendan! I opted for adding a command that presses keys and then waits for the value to change appropriately. But this new command will strip out the "\n" from the input string. Lmk what you think.

Reply:
After discussion online, I updated the step to wait until element's value changes to expected value.